docs/README.md

```diff
@@ -296,7 +296,7 @@ To add a new input test with a sample P4 code file (under `testdata/p4_16_sample
   * Use `BUG_CHECK()` instead of `assert`, and always supply an
     informative error message.
 
-  * Use `::error()` and `::warning()` for error reporting. See the
+  * Use `::P4::error()` and `::P4::warning()` for error reporting. See the
```

Comment:
We probably don't need to recommend using the global prefix now, it is hightly unlikely there will be `P4::error` clash and if there is the programmer will probably be able to figure it out.

Reply:
How about changing them to `error()` and `warning()` then?

Reply:
I'm not sure why we ever recommended using the scope prefix -- in those odd cases where there would be a difference, you generally want the local one anyways (eg, in the TypeCheck pass).  The only place you need the scope qualifier is when a local error routine is calling the global one.

Reply:
I would probably say – whoever would need explicit scope qualifier should already know what they are doing. Anyone else should just use plain version. Especially in `P4` namespace.

Reply:
> How about changing them to `error()` and `warning()` then?

Done.